Description: meson 0.58 requires the env list to be flat, it's no longer
allowed to mix lists with scalar items.
diff --git a/test/automated/displayless/meson.build b/test/automated/displayless/meson.build
index 64b23bd..d7cf5db 100644
--- a/test/automated/displayless/meson.build
+++ b/test/automated/displayless/meson.build
@@ -45,8 +45,8 @@ foreach t: tests
   test(
     t[0],
     executable(t[0], t[1], files('test-utilities.c'), dependencies: libnautilus_dep),
-    env: [
-      test_env,
+    env: test_env +
+    [
       'G_TEST_BUILDDIR=@0@'.format(meson.current_build_dir()),
       'G_TEST_SRCDIR=@0@'.format(meson.current_source_dir())
     ],
@@ -64,8 +64,8 @@ foreach t: tracker_tests
     t[0],
     tracker_sandbox,
     args: ['--store-tmpdir', '--index-recursive-tmpdir', test_exe],
-    env: [
-      test_env,
+    env: test_env +
+    [
       'G_TEST_BUILDDIR=@0@'.format(meson.current_build_dir()),
       'G_TEST_SRCDIR=@0@'.format(meson.current_source_dir())
     ],
